### PR TITLE
WIP: Bastion floating on tide

### PR DESCRIPTION
**Pull request by o0Ignition0o**

### CODE REVIEW

File: _posts/2020-02-15-bastion-floating-on-tide.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+$ cargo install cargo-edit
+```
+
+Let’s start with a tide example:
+
+```bash
+$ cargo new bastion_with_tide # Create a new project
+$ cd bastion_with_tide # Go to the project folder
+$ cargo add async-std # Add async-std
+$cargo add tide # Add tide
```
Comment (skade) [inline _posts/2020-02-15-bastion-floating-on-tide.md]: Space missing

**Comment by o0Ignition0o** [comment]
Swapped the example for a more interesting one, as suggested by @skade <3
I ll hopefully soon add the bastion part, as well as an introduction and a conclusion !

**Comment by skade** [comment]
Still says fibonacci quite a bit ;)

**Comment by o0Ignition0o** [comment]
> Still says fibonacci quite a bit ;)

Oh :/ I tried to keep it mentionned so I can let readers head over to the Tide examples if/once https://github.com/http-rs/tide/pull/406 is merged

I can however remove it if need be ^^

File: _posts/2020-02-15-bastion-floating-on-tide.md [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+If this exercise doesn't ring a bell, don't worry, I didn't know about it either, and I had an amazing journey discovering Programming Erlang, by [Joe Armstrong](https://en.wikipedia.org/wiki/Joe_Armstrong_(programmer)).
+
+If you were interested in a simple fibonacci service using tide, head over to [the tide examples](https://github.com/http-rs/tide/tree/master/examples), it will probably be available by the time this article is released.
+
+Our prime number service will, given a number of digits, return a prime number with the expected number of digits.
+
+In pseudocode it would look like this:
+
+```
+GET: /prime/digits => return prime_number(digits)
```
Comment (felipesere) [inline _posts/2020-02-15-bastion-floating-on-tide.md]: maybe `/prime/:digits` I think the `:foo` notation is fairly common for "URL parameters"?

File: _posts/2020-02-15-bastion-floating-on-tide.md [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+
+Our prime number service will, given a number of digits, return a prime number with the expected number of digits.
+
+In pseudocode it would look like this:
+
+```
+GET: /prime/digits => return prime_number(digits) 
+```
+
+First of all, we want to extract some data from the route we create. We can leverage the powers of tide and get named parameters by specifying a route and a handler function:
```
Comment (felipesere) [inline _posts/2020-02-15-bastion-floating-on-tide.md]: power

File: _posts/2020-02-15-bastion-floating-on-tide.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+```rust
+        app.at(“/prime/:digits).get(prime_number);
+```
+
+The Tide router will extract d as a parameter, which will be queryable in our prime_number function:
```
Comment (felipesere) [inline _posts/2020-02-15-bastion-floating-on-tide.md]: Sometimes you write `Tide` and sometimes `tide`. I gut feel is that uppercase is more correct. 🤗

**Comment by o0Ignition0o** [comment]
Wow thanks a lot for taking time to review the WIP article, @felipesere I really appreciate it!
I will try to address the comments tomorrow :)

File: _posts/2020-02-23-bastion-floating-on-tide-part-1.md [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+
+### Add a little twist
+
+The more complex a system becomes, the more likely it is to fail. In order to introduce Bastion and a couple of its basic features, we will introduce a failure mode that will happen randomly in our example, and see how bastion can help us reason about failures and complex systems. But first, a disclaimer.
+
+### Disclaimer
+
+THIS EXAMPLE COMES AS IS... Yeah no that's not what I meant by a disclaimer.
+
+async-rs and the async-std library allow you to achieve a lot, and by a lot I really mean A LOT. The twist I'm about to introduce (leading to a panic) is _not _ idiomatic at all, and you're much better off leveraging Result, Option and other awesome Rust structures that were built to do this. The purpose of this example is not to make you think Bastion can solve this problem and async-rs cannot. async-rs can totally handle failures (and generating prime numbers for that matter) in a graceful way.
```
Comment (vertexclique) [inline _posts/2020-02-23-bastion-floating-on-tide-part-1.md]:
```suggestion
async-rs and the async-std library allow you to achieve a lot, and by a lot I really mean A LOT. The twist I'm about to introduce (leading to a panic) is _not_ idiomatic at all, and you're much better off leveraging Result, Option and other awesome Rust structures that were built to do this. The purpose of this example is not to make you think Bastion can solve this problem and async-rs cannot. async-rs can totally handle failures (and generating prime numbers for that matter) in a graceful way.
```

File: _posts/2020-02-23-bastion-floating-on-tide-part-1.md [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+Well maybe not… We're going to create a couple of functions that will allow us to compute a prime number with the correct width. 
+Long story short we will iterate over numbers with the correct number of digits starting from a random one, and use [a test to find out if the random number is a prime number](https://en.wikipedia.org/wiki/Primality_test#Pseudocode). It will look like this:
+
+```rust
+fn get_prime(num_digits: usize) -> u128 {
+    let min_bound = get_min_bound(num_digits);
+    // with num_digits = 4, max_bound == 10000
+    let max_bound = get_max_bound(num_digits);
+    // maybe_prime is a number in range [1000, 10000)
+    // the closing parenthesiss means it won't reach the number.
```
Comment (vertexclique) [inline _posts/2020-02-23-bastion-floating-on-tide-part-1.md]:
```suggestion
    // the closing parentheses means it won't reach the number.
```

**Pull request merged**